src/components/apply/menu.tsx

```diff
@@ -1,26 +1,153 @@
 import { applySteps, type ApplyStep } from "~/constants/apply";
 import { Button } from "../ui/button";
 import Link from "next/link";
+import { useMemo } from "react";
+import { getNextStep, getPreviousStep, getStepIndex } from "./navigation";
+import { useRouter } from "next/router";
+import {
+  Drawer,
+  DrawerClose,
+  DrawerContent,
+  DrawerDescription,
+  DrawerFooter,
+  DrawerHeader,
+  DrawerTitle,
+  DrawerTrigger,
+} from "../ui/drawer";
 
 type ApplyMenuProps = {
   step: ApplyStep | null;
 };
 
 export function ApplyMenu({ step }: ApplyMenuProps) {
+  const stepName = step && step?.charAt(0).toUpperCase() + step?.slice(1);
+  const isMobile = typeof window !== "undefined" && window.innerWidth < 640;
+
+  const stepIndex = useMemo(() => getStepIndex(step), [step]);
+  const prevStep = getPreviousStep(stepIndex);
+  const nextStep = getNextStep(stepIndex);
+
+  const router = useRouter();
+
   return (
-    // TODO: make this responsive
-    <div className="flex w-full justify-center gap-3 border-[1px] border-t-primary-300 bg-violet-100 py-2">
-      {applySteps.map((s) => (
-        <Button
-          key={s.step}
-          variant={s.step === step ? "apply" : "apply-ghost"}
-          asChild
+    <div
+      className={`fixed bottom-0 mx-auto h-fit w-screen justify-center gap-3 border-[1px] border-t-primary-300 bg-violet-100 py-2 sm:static`}
+    >
+      {isMobile && (
+        <Drawer>
```

Comment:
why can't we do className="sm:invisible" here or at the  as well? instead of having to compute isMobile in a weird way.

Reply:
the menu shouldn't show up on mobile at all, the styling is from before i was using drawer

Reply:
setting something to invisible still renders it so it'll still take up space so i prefer to not render at all if mobile (the cutoff for `isMobile` is the same as the tailwind sm cutoff so there's no inconsistency there either)

Reply:
wait can't you use hidden then? it's just that I think u should prefer to use css rather than js for layout stuff

Reply:
valid i forogot

Reply:
ok that's fine